### PR TITLE
ROS mission tests simple markdown documentation

### PR DESCRIPTION
**Pull request by dagar**
Not that useful yet.
https://github.com/PX4/Firmware/blob/5a1d98015040c0149a529745da2c606ad42c128d/integrationtests/python_src/px4_it/mavros/tests.md

TODO
 - each Jenkins build should either include the description or link to this (both Jenkins and posted flight review)
 - missions need meaningful naming
 - test everything!

### CODE REVIEW

File: integrationtests/python_src/px4_it/mavros/tests.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+
+
+
+
+
+| Test Name  | Description | Image |
+| ------------- | ------------- | ------------- |
+| [multirotor_box.mission](multirotor_box.mission "multirotor box") | Simple multirotor box mission. [Example](https://logs.px4.io/plot_app?log=c950b7f7-86cb-4b89-8674-b0f46265b823) | ![multirotor_box](images/multirotor_box.png) |
+| Offboard Attitude | Offboard attitude test with ROS. [Example](https://logs.px4.io/plot_app?log=17a5dbd3-c1a8-4d2e-97e5-bf81de048baf) | ![offboard_attitude](images/mc_offboard_attitude.png) |
```
Comment (hamishwillee) [inline integrationtests/python_src/px4_it/mavros/tests.md]: No link for offboard attitude/position tests?

Comment (dagar) [reply integrationtests/python_src/px4_it/mavros/tests.md]: Those are separate little python (rospy) tests. https://github.com/PX4/Firmware/blob/master/integrationtests/python_src/px4_it/mavros/mavros_offboard_attctl_test.py

Comment (hamishwillee) [reply integrationtests/python_src/px4_it/mavros/tests.md]: Consider simplest format for markdown table - omit the outer pipes, use just three dashes to mark columns. IMO makes things more readable. 

```
Test Name  | Description | Image 
 --- | --- | ---
[multirotor_box.mission](multirotor_box.mission "multirotor box") | Simple multirotor box mission. [Example](https://logs.px4.io/plot_app?log=c950b7f7-86cb-4b89-8674-b0f46265b823) | ![multirotor_box](images/multirotor_box.png)
Offboard Attitude | Offboard attitude test with ROS. [Example](https://logs.px4.io/plot_app?log=17a5dbd3-c1a8-4d2e-97e5-bf81de048baf) | ![offboard_attitude](images/mc_offboard_attitude.png)

Comment (hamishwillee) [reply integrationtests/python_src/px4_it/mavros/tests.md]: To link to individual tests you will need to add an id for each test. You can do that with html spans. `<span id="example_whatever"></span>`

**Comment by hamishwillee** [comment]
Some thoughts:
1. Purple for the mission path is not very easy to see. Offboard tracks much easier to see. If possible use more distinctive colour. Maybe also crop closer to the actual path. Maybe use a higher resolution image.
2. Is it worth moving the content to README in the directory? 
   - My understanding is that this will then render the tests on the page that gets displayed when you go to the parent directory. That is more discoverable.
   - Also allows you to have more general docs about what is in the folder.

**Comment by dagar** [comment]
1. These are screenshots of flight review, but I agree. If we wanted to get really fancy they should be rendered from the example URL.

**Comment by hamishwillee** [comment]
> These are screenshots of flight review, but I agree. 

OK, well just for comparison, screenshot below is 1200Kb, while yours are 200Kb. Somewhere in between would be good. 

image

Another idea, use the map view:

image

File: integrationtests/python_src/px4_it/mavros/tests.md (hunk 1 above)
Comment (lamping7) [reply integrationtests/python_src/px4_it/mavros/tests.md]: Links aren't need for these tests, but we should probably say what they are doing... 

That leads to an issue with all the tests. e.g. new_1 and new_2 are the essentially the same, but new_2 has a WP in the middle and the land location is different. If we have new_2, why do we need new_1? Does the landing location mean anything that makes it different from the other that requires a separate test. Another example is: old_1 and old_2 contain the same commands with different locations. It is commented in these what the locations are -- what it is supposed to do. 

This is what we need in the description column. Of course, before this, we need to decide what we want to test and create more tests. But, I'd forget about these initially and first come up with descriptions for tests we think we need. Then we can look at the tests we have, see if they fit the bill, or make new ones -- using QGC .plan format.

Comment (dagar) [reply integrationtests/python_src/px4_it/mavros/tests.md]: Yes, the current ones need to be tossed. I just want to get all the pieces in place (test definition, jenkins portions, basic description) so we can bring up a useful level of testing. I also think it's very important for what Jenkins posts to flight review to either link to this documentation or include it directly.

**Comment by lamping7** [comment]
1. I think the photo should match what will come up in flight review -- same map -- to compare easier when necessary. If the missions will change down the line, it might be easier (in terms of future maintenance) to render them from the example.
2. I don't have a strong opinion where this goes, but what about here?: https://dev.px4.io/en/test_and_ci/integration_testing.html (which recently became outdated as well)

File: integrationtests/python_src/px4_it/mavros/tests.md (hunk 1 above)
Comment (lamping7) [reply integrationtests/python_src/px4_it/mavros/tests.md]: I think a link is good idea to quickly see what it should be doing and look like on the map. Putting it in the feedback section should work fine.

**Comment by hamishwillee** [comment]
> I don't have a strong opinion where this goes, but what about here?: https://dev.px4.io/en/test_and_ci/integration_testing.html (which recently became outdated as well)

I discussed this with @dagar offline. I had a slight preference for that location too, but it doesn't matter as long as we have clear links between the places. I also thought having tests in README.md better than tests.md, so that they auto render when you go to the folder.

2. Importing the map would be cool, but not particularly easy. 

3. How about we make the image a link to the example, and change the row heading to Example (Click Image)?

**Comment by lamping7** [comment]
Because this will be under active development it makes sense to leave it on the firmware side rather than the dev guide -- so pull requests can include both -- and link to it in the dev guide.

Regarding the readme: You're probably right, although I really don't see users caring about development testing. Again no preference. Renaming it is fine.

2. I agree.

3. Removing the example link from the description to the image along with changing the column heading to reflect it is a good idea

**Comment by hamishwillee** [comment]
@dagar What is blocking moving this forward? I'm quite tempted to pull this into the devguide instead.

**Pull request closed without merging**